drivers/axi_core/axi_dac_core/axi_dac_core.c

```diff
@@ -952,15 +952,16 @@ int32_t axi_dac_init(struct axi_dac **dac_core,
 		      AXI_DAC_MMCM_RSTN | AXI_DAC_RSTN);
 
 	axi_dac_write(dac, AXI_DAC_REG_RATECNTRL, AXI_DAC_RATE(init->rate));
-	axi_dac_data_setup(dac);
-	axi_dac_write(dac, AXI_DAC_REG_SYNC_CONTROL, AXI_DAC_SYNC);
 
 	no_os_mdelay(100);
 
 	ret = axi_dac_init_finish(dac);
 	if (ret)
 		goto error;
 
+	axi_dac_data_setup(dac);
```

Comment:
unrelated to your PR but we might have this bug elsewhere, it's worth recursively searching for axi_dac_init_finish and see if the setup/write are done before :D